### DISCUSSION TITLE
How to live preview php?

### DISCUSSION THREAD

**Discussion by europacafe** [coder/code-server#6687] [Q&A]
I run code-server docker container on my unraid. Live view or live preview extension works great with html code. For PHP, I install php8.1 inside the code-server container, but still can't get the php code executed.
![image](https://github.com/coder/code-server/assets/51871976/791cf4f0-ab2c-4483-ae70-2693b8a9cf96)

How can I do the same for my php code?

**Comment by code-asher** [comment]
Is that the whole error?  Can you run that command in the terminal manually and see if it also fails?  What is the ID of the extension you are using?

**Reply by europacafe** [reply]
I ran it on terminal (inside container console, or code-server terminal, it was just quiet and returned to prompt without any reported error.

**Reply by europacafe** [reply]
If I create a new folder and a simple php file inside the code-server, it executes properly.
For my other working php websites I uploaded to code-server, none is working with similar error. The php is actually execute those php files, but always get errors.

**Reply by code-asher** [reply]
Strange, I was not able to reproduce any issues but I might be using the wrong plugin.  If you can make a minimum reproduction I will try running it, but otherwise it sounds like it might be a problem separate from code-server.

**Reply by europacafe** [reply]
Could you elaborate how you set up your code-server, extensionห you use with their settings. How you installed php, inside the container or a separate php docker, etc.. I've been trying it the whole week. I also installed most php extensions/utilities inside my code-server container...

**Reply by code-asher** [reply]
I am only running a simple PHP file, which you said works for you as well, right?  So I am not sure my setup will give you anything new.

But I am running everything in a single Ubuntu Docker container.

In a terminal:

```
$ sudo apt install php
$ mkdir php-test
$ echo "hello world" > test/index.php
```

Then in code-server:

1. File > Open Folder > navigate to php-test
2. Extensions > search for "php live" > install `yandeu.five-server`
3. Open settings > set `fileServer.php.executable` to `/usr/bin/php`
4. Open `index.php`
5. Command palette > run "Open with Five Server"
6. Click "open in browser" from popup

I am not sure why the setting is required in my case, for some reason it does not find php in my path.  But you mentioned yours does run php, it just errors, so it might not fix anything for you.

----

Also, PHP has a built-in web server for development so you could just skip the plugin and run this in the terminal:

```
php -S localhost:8000
```

And then click the "open in browser" button in the popup that appears.

**Reply by europacafe** [reply]
Thanks for sharing your detailed settings. I've done exactly the same, but I don't even need to specify the php executable path to run a simple php.
When I run php -S .... command as you suggested, I saw on it saying, and no "open in browser" button
`PHP 8.1.2-1ubuntu2.14 Development Server (http://localhost:8000) started`
The other thing, when I run "Open with Five Server", it will display a link in console, I then press ctrl+mouse click to open it in browser. However, my browser've never open the link, instead it always opens just the root proxy link, i.e., 192.168.2.93:8443/proxy/5500/, instead of opening the subfolder directly (https://192.168.2.93:8443/proxy/5500/test/index.php,) even though I installed the live server extension and specified both actual server addresses (https://192.168.2.93:8443/proxy/5500/) and live server address (http:127.0.0.1:5500)
Would you mind if I sent you my whole website folders/files, to test and prove whether it is my setup issues or my website itself.

**Reply by code-asher** [reply]
Sure, send them over and I will give it a test.
